core/src/main/java/com/sequenceiq/cloudbreak/service/image/ImageType.java

```diff
@@ -0,0 +1,31 @@
+package com.sequenceiq.cloudbreak.service.image;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableMap.Builder;
+
+public enum ImageType {
+    FREEIPA,
+    DATAHUB,
+    DATALAKE,
+    INVALID;
+
+    private static final ImmutableMap<String, ImageType> VALUEMAP;
+
+    static {
```

Comment:
We have a generic implementation for this: DefaultEnumConverter
You should create a converter and reference it in the entity as a @Converter so hibernate will know how to map String to ImageType and vice versa. Sample what you can use as a pattern:  CommonStatusConverter

Reply:
ImageTypeConverter is used for the conversation.